examples/custom-service.mdx

```diff
@@ -20,71 +38,163 @@ edition = "2021"
 [lib]
 
 [dependencies]
-shuttle-service = { version = "0.9.0" }
-shuttle-shared-db = { version = "0.9.0", features = ["postgres"] }
-sqlx = { version = "0.6", features = ["runtime-tokio-native-tls", "postgres"] }
+anyhow = "1.0.62"
+axum = "0.6.4"
+hyper = "0.14.24"
+poise = "0.5.2"
+serde = "1.0"
+shuttle-secrets = "0.9.0"
+shuttle-service = { version = "0.9.0"}
+sync_wrapper = "0.1.2"
+tokio = "1.22.0"
 ```
 
-This section will explain how to create a custom shuttle service. To do this we need 
-to return a struct from our `shuttle_service::main` function that implements `shuttle_service::Service`.
-
-In this simple example we will implement `Service` for a `PgPool`, that will simply return 'Hello world'
-from the database on startup. In Rust we can't implement foreign traits for foreign types, so we have 
-to create a wrapper around our `PgPool`.
+### Getting Started
+To get started, we need to return a wrapper struct from our `shuttle_service::main` function that 
+implements `shuttle_service::Service`, which we can declare like so:
 
-```rust
-struct PoolService {
-    pool: sqlx::PgPool,
+```rust lib.rs
+pub struct CustomService {
+    discord_bot:
+        FrameworkBuilder<Data, Box<(dyn std::error::Error +
+         std::marker::Send + Sync + 'static)>>,
```

Comment:
Can the error type here just be shuttle_service::CustomError?

Reply:
As far as I've experimented, I don't think so?

The FrameworkBuilder type itself expects something that uses `Box<(dyn std::error::Error + std::marker::Send + Sync + 'static)>` for an error type - it looks like trying to put in `shuttle_service::CustomError` doesn't satisfy the type checking.

[Relevant repo link with code](https://github.com/joshua-mo-143/axum-with-discord-bot)